File: snippets/dart_snippets/lib/list_payments.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';

Future<Payment?> getPayment() async {
  // ANCHOR: get-payment
  String paymentHash = "<payment hash>";
  GetPaymentRequest req = GetPaymentRequest.lightning(paymentHash: paymentHash);
  Payment? payment = await breezSDKLiquid.instance!.getPayment(req: req);
  // ANCHOR_END: get-payment
  return payment;
}

Future<List<Payment>> listPayments() async {
  // ANCHOR: list-payments
  ListPaymentsRequest req = ListPaymentsRequest();
  List<Payment> paymentsList = await breezSDKLiquid.instance!.listPayments(req: req);
  // ANCHOR_END: list-payments
  return paymentsList;
}

Future<List<Payment>> listPaymentsFiltered() async {
  // ANCHOR: list-payments-filtered
  ListPaymentsRequest req = ListPaymentsRequest(
    filters: [PaymentType.send],
    fromTimestamp: 1696880000,
    toTimestamp: 1696959200,
    offset: 0,
    limit: 50,
  );
  List<Payment> paymentsList = await breezSDKLiquid.instance!.listPayments(req: req);
  // ANCHOR_END: list-payments-filtered
  return paymentsList;
}

Future<List<Payment>> listPaymentsDetailsAddress() async {
  // ANCHOR: list-payments-details-address
  String address = "<Bitcoin address>";
  ListPaymentsRequest req = ListPaymentsRequest(
    details: ListPaymentDetails_Bitcoin(address: address),
  );
  List<Payment> paymentsList = await breezSDKLiquid.instance!.listPayments(req: req);
  // ANCHOR_END: list-payments-details-address
  return paymentsList;
}

Future<List<Payment>> listPaymentsDetailsDestination() async {
  // ANCHOR: list-payments-details-destination
  String destination = "<Liquid BIP21 or address>";
  ListPaymentsRequest req = ListPaymentsRequest(
    details: ListPaymentDetails_Liquid(destination: destination),
  );
  List<Payment> paymentsList = await breezSDKLiquid.instance!.listPayments(req: req);
  // ANCHOR_END: list-payments-details-destination
  return paymentsList;
}

File: snippets/react-native/list_payments.ts
import {
  BindingLiquidSdk,
  GetPaymentRequest,
  ListPaymentDetails,
  PaymentType
} from '@breeztech/breez-sdk-liquid-react-native'

const exampleGetPayment = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: get-payment
  const paymentHash = '<payment hash>'
  const paymentByHash = sdk.getPayment(new GetPaymentRequest.PaymentHash({ paymentHash }))

  const swapId = '<swap id>'
  const paymentBySwapId = sdk.getPayment(new GetPaymentRequest.SwapId({ swapId }))
  // ANCHOR_END: get-payment
}

const exampleListPayments = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-payments
  const payments = sdk.listPayments({
    filters: undefined,
    states: undefined,
    fromTimestamp: undefined,
    toTimestamp: undefined,
    offset: undefined,
    limit: undefined,
    details: undefined,
    sortAscending: undefined
  })
  // ANCHOR_END: list-payments
}

const exampleListPaymentsFiltered = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-payments-filtered
  try {
    const payments = sdk.listPayments({
      filters: [PaymentType.Send],
      fromTimestamp: BigInt(1696880000),
      toTimestamp: BigInt(1696959200),
      offset: 0,
      limit: 50,
      states: undefined,
      details: undefined,
      sortAscending: undefined
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-payments-filtered
}

const exampleListPaymentsDetailsAddress = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-payments-details-address
  try {
    const payments = sdk.listPayments({
      details: new ListPaymentDetails.Bitcoin({ address: '<Bitcoin address>' }),
      filters: undefined,
      states: undefined,
      fromTimestamp: undefined,
      toTimestamp: undefined,
      offset: undefined,
      limit: undefined,
      sortAscending: undefined
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-payments-details-address
}

const exampleListPaymentsDetailsDestination = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-payments-details-destination
  try {
    const payments = sdk.listPayments({
      details: new ListPaymentDetails.Liquid({ assetId: undefined, destination: '<Liquid BIP21 or address>' }),
      filters: undefined,
      states: undefined,
      fromTimestamp: undefined,
      toTimestamp: undefined,
      offset: undefined,
      limit: undefined,
      sortAscending: undefined
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-payments-details-destination
}
